darts/models/forecasting/arima.py

```diff
@@ -107,6 +125,15 @@ def _predict(
                 exog=future_covariates.values() if future_covariates else None,
             )
 
+        # restoring statsmodels results object state
```

Comment:
Why is this needed?

Reply:
The default behavior of the `predict()` method, without specifying any `series`, forecasts the next `n` steps of the training series. To preserve this behavior also after calling `predict` with a different TS (which changes the state of the underlying `ARIMAResults` object calling `apply()`), I call `apply` again with the training data.